File: leetcode/pascals_triangle_2/pascals_triangle_2.py
class Solution:
	def getRow(self, rowIndex: int) -> List[int]:
		if (rowIndex == 0):
			return [1]
		elif (rowIndex == 1):
			return [1,1]

		last = [1,1]

		for i in range(2, rowIndex+1):
			next = [1]
			for u in range(1, i+1):
				if (u <= i//2):
					next += [last[u] + last[u-1]]
				else:
					next += [next[i - u]]
			last = next
		
		return next

File: leetcode/pascals_triangle_2/pascals_triangle_2.cpp
class Solution {
public:
	vector<int> getRow(int rowIndex) {
		vector<int> last{1};
		vector<int> next = last;

		for (int i = 1; i <= rowIndex; i++) {
			next = {1};
			for (int u = 1; u <= i; u++) {
				if (u <= i/2)
					next.push_back(last.at(u) + last.at(u-1));
				else
					next.push_back(next.at(i - u));
			}
			last = next;
		}
		return next;
	}
};  
